src/CdpRegistry.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.6.12;
+
+interface CdpManagerLike {
+    function open(bytes32, address) external returns (uint256);
+}
+
+contract CdpRegistry {
+    mapping (uint256 => address) public owns; // CDPId => Owner
+    mapping (uint256 => bytes32) public ilks; // CDPId => Ilk
+    mapping (bytes32 => mapping (address => uint256)) public cdps; // Ilk => Owner => CDPId
+
+    address public immutable cdpManager;
+
+    event NewCdpRegistered(address indexed sender, address indexed owner, uint256 indexed cdp);
+
+    constructor(address cdpManager_) public {
+        cdpManager = cdpManager_;
+    }
+
+    function open(
+        bytes32 ilk,
+        address usr
```

Comment:
Wondering if we should just allow the `msg.sender` to create its CDP. As you can't have multiple ones you will be forced to have the id that whoever can create for you.

Reply:
Yeah, was to trying to keep the same interface as the original dss-proy-actions.
I think we might want to just require that `usr` is `msg.sender`, so we can stay with the same interface if we upgrade.
This is maybe what you meant as well.

Reply:
Btw @gbalabasquer you meant we should just allow `address(this)` right? not `msg.sender`.

Reply:
I mean that `usr` should be `msg.sender`, exactly as you described.
Doing the require to keep the same interface sounds good to me.

Reply:
Ok, added that.

Reply:
Looks good.